src/content/docs/en/guides/middleware.mdx

```diff
@@ -200,3 +192,7 @@ greeting response
 auth response
 validation response
 ```
+
+### Handling 404 pages
+
+Middleware will run for an on-demand rendered route while Astro attempts to render a 404 page. However, it is up to the [adapter](/en/guides/server-side-rendering/) to decide whether that code runs. Some adapters may serve a platform-specific error page instead.
```

Comment:
```suggestion
Middleware will run for an on-demand rendered route while Astro attempts to render a 404 page. However, it is up to the [adapter](/en/guides/server-side-rendering/) to decide whether that code runs. Some adapters may serve a platform-specific error page instead.

If a page fails to render due to an error, astro will attempt to render your custom 500 page. If the error page is set to render on-demand, the middleware will run a second time for that request.
```

Reply:
I'm trying to figure out how *important* it is to add these two paragraphs. 500 is generic error, right? Does it *matter* that middleware runs if the page 500s? What practical value is this to the author, and would they change anything about how they write their code knowing this? In which situations would it make a difference?

Similarly, if a 500 page is being served,how much does it *matter* that they don't have `Astro.locals`? 

I'm going by the "not everything is true belongs in docs" principle, and want to include things that will affect people's decisions, and that they need to know to build properly. I'm going to need your help to figure out if these fit that criteria.

Reply:
Not that important.

The most important point would be that 500 page could miss `locals` _sometimes_. However, even that I don't feel too strongly about - it came up only once.